### PR TITLE
4.0.0-alpha.31

### PR DESCRIPTION
**Pull request by DanailH**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,57 @@
...
+## [4.0.0-alpha.31](https://github.com/mui-org/material-ui-x/compare/v4.0.0-alpha.30...v4.0.0-alpha.31)
+
+_June 9, 2021_
+
+Big thanks to the 6 contributors who made this release possible. Here are some highlights ✨:
+
+- 💅 Allow to customize GridToolbarExport's CSV export (#1695) @michallukowski
+- 🐛 Allow to deselect rows with CTRL + click (#1813) @ZeeshanTamboli
+- ⚡️ Refactor scroll size detector (#1703) @dtassone
+- 🕹 Automatically generate API docs (#1529) @m4theushw
```
Comment (m4theushw) [inline CHANGELOG.md:15]:
```suggestion
- 📖 Add [documentation](https://material-ui.com/api/data-grid/) for interfaces and events (#1529) @m4theushw
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -3,6 +3,57 @@
...
 See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.
 
+## [4.0.0-alpha.31](https://github.com/mui-org/material-ui-x/compare/v4.0.0-alpha.30...v4.0.0-alpha.31)
+
+_June 9, 2021_
+
+Big thanks to the 6 contributors who made this release possible. Here are some highlights ✨:
+
+- 💅 Allow to customize GridToolbarExport's CSV export (#1695) @michallukowski
+- 🐛 Allow to deselect rows with CTRL + click (#1813) @ZeeshanTamboli
```
Comment (m4theushw) [inline CHANGELOG.md:13]:
```suggestion
- 🐛 Allow to deselect rows with <kbd>CTRL</kbd> + click (#1813) @ZeeshanTamboli
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -3,6 +3,57 @@
...
+
+```diff
+-cellClassName?: string | string[] | (params: GridCellParams) => string;
++cellClassName?: string | (params: GridCellParams) => string;
+```
+
+```diff
+-headerClassName?: string | string[];
++headerClassName?: string | (params: GridColumnHeaderParams) => string;
+```
```
Comment (m4theushw) [inline CHANGELOG.md:32]: Indentation?

File: CHANGELOG.md [hunk 4]
```diff
@@ -3,6 +3,57 @@
...
+
+#### Changes
+
+- [DataGrid] Add `valueParser` to parse values entered by the user (#1785) @m4theushw
+- [DataGrid] Allow to customize GridToolbarExport's CSV export (#1695) @michallukowski
+- [DataGrid] Allow to deselect rows with CTRL + click (#1813) @ZeeshanTamboli
+- [DataGrid] Exploring Plugin api (#1720) @dtassone
+- [DataGrid] Fix cell height after changing grids density (#1819) @DanailH
```
Comment (m4theushw) [inline CHANGELOG.md:40]:
```suggestion
- [DataGrid] Fix cell height after changing grid density (#1819) @DanailH
```

File: CHANGELOG.md [hunk 5]
```diff
@@ -3,6 +3,57 @@
...
+```diff
+-headerClassName?: string | string[];
++headerClassName?: string | (params: GridColumnHeaderParams) => string;
+```
+
+#### Changes
+
+- [DataGrid] Add `valueParser` to parse values entered by the user (#1785) @m4theushw
+- [DataGrid] Allow to customize GridToolbarExport's CSV export (#1695) @michallukowski
+- [DataGrid] Allow to deselect rows with CTRL + click (#1813) @ZeeshanTamboli
```
Comment (m4theushw) [inline CHANGELOG.md:38]:
```suggestion
- [DataGrid] Allow to deselect rows with <kbd>CTRL</kbd> + click (#1813) @ZeeshanTamboli
```

File: CHANGELOG.md [hunk 6]
```diff
@@ -3,6 +3,57 @@
...
+- [DataGrid] Exploring Plugin api (#1720) @dtassone
```
Comment (m4theushw) [inline CHANGELOG.md:39]: I'm open for suggestions. 😁

```suggestion
- [DataGrid] Improve general architecture to better isolate hooks (#1720) @dtassone
```

Comment (DanailH) [reply CHANGELOG.md:40]: Not sure. I'll merge this one. Let's see if someone else has a different idea.

**Review by m4theushw — APPROVED** [review]

**Review by dtassone — APPROVED** [review]

**Comment by oliviertassinari** [comment]
Look great, nice polish @m4theushw

**Pull request merged**